yarn-project/simulator/src/acvm/acvm.ts

```diff
@@ -105,7 +105,7 @@ export async function acvm(
         }
 
         const result = await oracleFunction.call(callback, ...args);
-        return [result];
+        return Array.isArray(result) && result.length === 0 ? [] : [result];
```

Comment:
Without this change, ACVM complained with `Assertion failed: 1 output values were provided as a foreign call result for 0 destination slots`

Reply:
Hmm seems like this breaks other calls:

```
          +e2e-tests *failed* |     Simulation error: Assertion failed: 0 output values were provided as a foreign call result for 1 destination slots 'call_public_function_oracle(
          +e2e-tests *failed* |             contract_address,
          +e2e-tests *failed* |             function_selector,
          +e2e-tests *failed* |             args_hash,
          +e2e-tests *failed* |             side_effect_counter,
          +e2e-tests *failed* |             is_static_call,
          +e2e-tests *failed* |             is_delegate_call
          +e2e-tests *failed* |         )'
```

Any tips on how to fix? I don't have a firm grasp on (ie I have no idea) how ACVM expects the callback return values.

Reply:
I think that our oracle resolver doesn't handle very well returning more/less than 1 result (that result being a value or an array).
ACVM oracles expect returns to be an array of ValueOrArray. So for example if you are returning an array of fields, the ACVM expects [array]. If you are returning a struct with a field and an array, it'd be [field, array]. I think our typed oracle stuff assumes always returning one item, it being a field or an array. 
I think it needs some updating!

Reply:
> I think our typed oracle stuff assumes always returning one item, it being a field or an array.

Got it, so the issue is exclusively on the side of aztec-packages then, right? I can give it a shot.